### ISSUE TITLE
Eager loading is not performed when creating or editing a new object.

### ISSUE THREAD

**Issue by spatarel** [activescaffold/active_scaffold#612]
I have a `:problems` column configured as such:

```
class Teachers::LessonsController < TeacherController
  active_scaffold :lesson do |conf|
    conf.columns[:problems].form_ui = :chosen
    conf.columns[:problems].includes = [problems: :site]
    conf.columns[:problems].options = {:include_blank => nil}
    conf.columns[:problems].clear_link
  end
end
```

I have a `:problem` model with a `to_label` function using the `:site` relation:
```
class Problem < ActiveRecord::Base
  belongs_to :site

  def to_label
    site.name + ': ' + slug + ' - ' + name
  end
end
```

When I try to create a new Lesson object (or edit an existing one), all the problems need to be loaded into a chosen object. This triggers a N + 1 query problem because the eager loading option is ignored.

More precisely, I can see this in the server output:
```
  CACHE Site Load (0.0ms)  SELECT  "sites".* FROM "sites" WHERE "sites"."id" = $1 LIMIT $2  [["id", 8], ["LIMIT", 1]]
  CACHE Site Load (0.0ms)  SELECT  "sites".* FROM "sites" WHERE "sites"."id" = $1 LIMIT $2  [["id", 8], ["LIMIT", 1]]and
  CACHE Site Load (0.0ms)  SELECT  "sites".* FROM "sites" WHERE "sites"."id" = $1 LIMIT $2  [["id", 8], ["LIMIT", 1]]
  CACHE Site Load (0.0ms)  SELECT  "sites".* FROM "sites" WHERE "sites"."id" = $1 LIMIT $2  [["id", 8], ["LIMIT", 1]]
  CACHE Site Load (0.1ms)  SELECT  "sites".* FROM "sites" WHERE "sites"."id" = $1 LIMIT $2  [["id", 8], ["LIMIT", 1]]
  CACHE Site Load (0.0ms)  SELECT  "sites".* FROM "sites" WHERE "sites"."id" = $1 LIMIT $2  [["id", 8], ["LIMIT", 1]]
  CACHE Site Load (0.0ms)  SELECT  "sites".* FROM "sites" WHERE "sites"."id" = $1 LIMIT $2  [["id", 8], ["LIMIT", 1]]
  CACHE Site Load (0.0ms)  SELECT  "sites".* FROM "sites" WHERE "sites"."id" = $1 LIMIT $2  [["id", 8], ["LIMIT", 1]]
```
for every problem (and there are a few thousands of them). As you can expect, this turns out to be extremely slow.

How do I correctly set up the eager loading?
Or is this a real issue?

**Comment by scambra** [comment]
includes option in association columns is eager loaded on forms only if sort is defined, in case is needed because is sorted by associated record. I'm adding to 3-5-stable because it's good idea.

Anyway, if you want to change query looking for association options for forms, you can override association_klass_scoped:

```ruby
module Teachers::LessonsHelper
  def association_klass_scoped(association, klass, record)
    if association.name == :problems
      super.includes(:site)
    else
      super
    end
  end
end
```

**Comment by scambra** [comment]
fixed by 93df8a7

**Comment by spatarel** [comment]
Thank you! It works like a charm! :)
